jvm/src/test/java/com/microsoft/bond/protocol/BinaryStreamReaderTest.java

```diff
@@ -0,0 +1,693 @@
+        }
+
+        // end of stream
+        try {
+            short v = r.readInt16();
+            fail("Operation can't succeed: " + v);
+        } catch (EOFException e) {
+            // success
+        } catch (IOException e) {
+            fail("Other IOException can't be thrown here: " + e);
+        }
+    }
+
+    @Test
+    public void readInt32() {
+        // setup
+        byte[] inputBytes = new byte[]{
+                // 0x7F32E35A
+                (byte) 0x5A, (byte) 0xE3, (byte) 0x32, (byte) 0x7F,
+
+                // 0xBB00AAC0
+                (byte) 0xC0, (byte) 0xAA, (byte) 0x00, (byte) 0xBB
+        };
+        ByteArrayInputStream bais = new ByteArrayInputStream(inputBytes);
+        BinaryStreamReader r = new BinaryStreamReader(bais);
+
+        // read first
+        try {
+            int v = r.readInt32();
+            assertEquals(0x7F32E35A, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read second
+        try {
+            int v = r.readInt32();
+            assertEquals(0xBB00AAC0, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // end of stream
+        try {
+            int v = r.readInt32();
+            fail("Operation can't succeed: " + v);
+        } catch (EOFException e) {
+            // success
+        } catch (IOException e) {
+            fail("Other IOException can't be thrown here: " + e);
+        }
+    }
+
+    @Test
+    public void readInt64() {
+        // setup
+        byte[] inputBytes = new byte[]{
+                // 0x325476987F32E35A
+                (byte) 0x5A, (byte) 0xE3, (byte) 0x32, (byte) 0x7F,
+                (byte) 0x98, (byte) 0x76, (byte) 0x54, (byte) 0x32,
+
+                // 0xCCFF3113BB00AAC0
+                (byte) 0xC0, (byte) 0xAA, (byte) 0x00, (byte) 0xBB,
+                (byte) 0x13, (byte) 0x31, (byte) 0xFF, (byte) 0xCC
+        };
+        ByteArrayInputStream bais = new ByteArrayInputStream(inputBytes);
+        BinaryStreamReader r = new BinaryStreamReader(bais);
+
+        // read first
+        try {
+            long v = r.readInt64();
+            assertEquals(0x325476987F32E35AL, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read second
+        try {
+            long v = r.readInt64();
+            assertEquals(0xCCFF3113BB00AAC0L, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // end of stream
+        try {
+            long v = r.readInt64();
+            fail("Operation can't succeed: " + v);
+        } catch (EOFException e) {
+            // success
+        } catch (IOException e) {
+            fail("Other IOException can't be thrown here: " + e);
+        }
+    }
+
+    @Test
+    public strictfp void readFloat() {
```

Comment:
I am not 100% sure but we do a bit of floating-point arithmetic (convert double to float). The strictfp makes sure that this behavior is consistent across all JVM implementations.

---
In reply to: [119250083](https://github.com/Microsoft/bond/pull/465#discussion_r119250083) [](ancestors = 119250083)